packages/forms/src/UIForm/fieldsets/CollapsibleFieldset/CollapsibleFieldset.component.test.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import cases from 'jest-in-case';
+import { shallow } from 'enzyme';
+import createCollapsibleFieldset from './CollapsibleFieldset.component';
+
+function customTitle(value, schema) {
+	return `${schema.title}: ${value.firstname} ${value.lastname}`;
+}
+
+const schema = {
+	title: 'Basic',
+	items: [
+		{
+			key: ['age'],
+			title: 'Age',
+			schema: { type: 'number' },
+			type: 'number',
+		},
+		{
+			key: ['address'],
+			title: 'Adress',
+			schema: { type: 'string' },
+			type: 'text',
+		},
+	],
+};
+
+const value = {
+	firstname: 'Jimmy',
+	lastname: 'Somsanith',
+};
+
+describe('CollapsibleFieldset', () => {
+	cases(
+		'should render',
+		opts => {
+			// given
+			const CollapsibleFieldset = createCollapsibleFieldset(opts.titleFn);
+
+			// when
+			const wrapper = shallow(
+				<CollapsibleFieldset
+					id={'my-fieldset'}
+					schema={schema}
+					value={{ ...value, isClosed: opts.isClosed }}
+				/>,
+			);
+
+			// then
+			expect(wrapper.getNode()).toMatchSnapshot();
+		},
+		{
+			'a full fieldset (header and body)': { isClosed: false },
+			'a collapsed fieldset (header only)': { isClosed: true },
+			'a custom title': { isClosed: false, titleFn: customTitle },
+		},
+	);
+
+	cases(
+		'should toggle',
+		opts => {
+			// given
+			const CollapsibleFieldset = createCollapsibleFieldset();
+			const onChange = jest.fn();
+			const event = {
+				stopPropagation: jest.fn(),
+				preventDefault: jest.fn(),
+			};
```

Comment:
Can't you extract all the options cases independent constant outside of the function block?

Reply:
They are only constants within the function block. It allows to not have interference between tests.
In the tests, I check if the function has been called. If not initialized on each test (so not constants), the assertion will be always true after the first test.

Reply:
mOK